src/gt4py/next/embedded/function_field.py

```diff
@@ -0,0 +1,235 @@
+        return self._binary_operation(operator.gt, other)
+
+    def __ge__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(operator.ge, other)
+
+    def __and__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(operator.and_, other)
+
+    def __or__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(operator.or_, other)
+
+    def __xor__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(operator.xor, other)
+
+    def __radd__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(lambda x, y: y + x, other)
+
+    def __rfloordiv__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(lambda x, y: y // x, other)
+
+    def __rmul__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(lambda x, y: y * x, other)
+
+    def __rsub__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(lambda x, y: y - x, other)
+
+    def __rtruediv__(self, other: common.Field | core_defs.ScalarT) -> common.Field:
+        return self._binary_operation(lambda x, y: y / x, other)
+
+    def __pos__(self) -> common.Field:
+        return self._unary_op(operator.pos)
+
+    def __neg__(self) -> common.Field:
+        return self._unary_op(operator.neg)
+
+    def __invert__(self) -> common.Field:
+        return self._unary_op(operator.invert)
+
+    def __abs__(self) -> common.Field:
+        return self._unary_op(abs)
+
+    def __call__(self, *args, **kwargs) -> common.Field:
+        return self.func(*args, **kwargs)
+
+    def remap(self, *args, **kwargs) -> common.Field:
+        raise NotImplementedError("Method remap not implemented")
+
+
+def _compose(operation: Callable, *fields: FunctionField) -> Callable:
+    return lambda *args: operation(*[f.func(*args) for f in fields])
+
+
+def _broadcast(field: FunctionField, dims: tuple[common.Dimension, ...]) -> FunctionField:
+    def broadcasted_func(*args: int):
+        selected_args = [args[i] for i, dim in enumerate(dims) if dim in field.domain.dims]
+        return field.func(*selected_args)
+
+    named_ranges = embedded_common._broadcast_domain(field, dims)
+    return FunctionField(broadcasted_func, common.Domain(*named_ranges), _skip_invariant=True)
+
+
+def _is_nd_array(other: Any) -> TypeGuard[nd._BaseNdArrayField]:
+    return isinstance(other, nd._BaseNdArrayField)
+
+
+def constant_field(
+    value: core_defs.ScalarT, domain: common.Domain = common.Domain()
+) -> common.Field:
+    return FunctionField(lambda *args: value, domain, _skip_invariant=True)
+
+
+FunctionField.register_builtin_func(fbuiltins.broadcast, _broadcast)
```

Comment:
to keep you busy: we should add all the functions from `fbuiltins` here as well

Reply:
should be done now